src/metadata.rs

```diff
@@ -61,9 +61,10 @@ impl MetadataStorage {
     where
         T: Any,
     {
-        self.entries
-            .remove(&TypeId::of::<T>())
-            .map(|meta| *(Box::<(dyn Any + 'static)>::downcast::<T>(meta).unwrap()))
+        self.entries.remove(&TypeId::of::<T>()).map(|meta| {
+            *(Box::<(dyn Any + 'static)>::downcast::<T>(meta)
+                .expect("the given type does not match the actual"))
+        })
```

Comment:
Shouldn't we also remove the expects in this file?

Reply:
Yes, I did it that way when native_assert still wasn't there, and I totally forgot of changing it. Thanks, I'll change it

Reply:
Gave it a second thought, we are having the same issue as in libfuncs. The error might happen inside the map, so the only solution to remove the panic would be to make it eager.